domains/transfer-agreements/TransferAgreements.API/API/Services/WalletDepositEndpointService.cs

```diff
@@ -56,23 +55,44 @@ public async Task<Guid> CreateReceiverDepositEndpoint(string bearerToken, string
         var walletServiceClient = new WalletService.WalletServiceClient(channel);
 
         var headers = new Metadata
-            { { "Authorization", $"Bearer {bearerToken}" } };
+        {
+            { "Authorization", bearerToken }
+        };
+
         var wde = Base64Converter.ConvertToWalletDepositEndpoint(base64EncodedWalletDepositEndpoint);
         var walletRequest = new CreateReceiverDepositEndpointRequest
         {
             Reference = receiverTin,
             WalletDepositEndpoint = wde
         };
+        try
+        {
+            var response = await walletServiceClient.CreateReceiverDepositEndpointAsync(walletRequest, headers);
+            Guid receiverReference = new(response.ReceiverId.Value);
 
-        var response = await walletServiceClient.CreateReceiverDepositEndpointAsync(walletRequest, headers);
-
-        Guid receiverReference = new(response.ReceiverId.Value);
+            if (receiverReference == Guid.Empty)
+            {
+                throw new InvalidOperationException("The receiver Id from the WalletService cannot be an empty Guid.");
+            }
 
-        if (receiverReference == Guid.Empty)
+            return receiverReference;
+        }
+        catch (Exception ex)
         {
-            throw new ArgumentException("The receiver Id cannot be an empty Guid.", nameof(response.ReceiverId));
+            HandleException(ex, "creating ReceiverDepositEndpoint");
+            throw;
         }
+    }
 
-        return receiverReference;
+    private void HandleException(Exception ex, string actionContext)
```

Comment:
Minor thing: In my book, this is not "handling" the exception, only logging it.

Reply:
I removed the method, as your point regarding the difference between exceptions were just the message template, thus rendering it obsolete.